src/registrar/models/dns/dns_zone.py

```diff
@@ -1,12 +1,52 @@
 from django.db import models
+
+from registrar.models.dns.dns_soa import DnsSoa
 from ..utility.time_stamped_model import TimeStampedModel
+from django.contrib.postgres.fields import ArrayField
 
 
 class DnsZone(TimeStampedModel):
+    class ZoneModes(models.TextChoices):
+        STANDARD = "standard", "Standard"
+        CDN_ONLY = "cdn_only", "CDN Only"
+        DNS_ONLY = "dns_only", "DNS Only"
+
     dns_account = models.ForeignKey("DnsAccount", on_delete=models.CASCADE, related_name="zones")
+
     vendor_dns_zone = models.ManyToManyField(
         "registrar.VendorDnsZone", through="DnsZone_VendorDnsZone", related_name="zones"
     )  # type: ignore
+
     domain = models.OneToOneField(
         "registrar.Domain", primary_key=True, null=False, on_delete=models.CASCADE
     )  # apex domain
+
+    soa = models.ForeignKey(
+        "registrar.DnsSoa", on_delete=models.SET_DEFAULT, related_name="+", default=DnsSoa.get_default_pk
```

Comment:
I'm not familiar with using "+" as a related name. Is that a convention? I would have used `zone` so that one could query with `dns_soa.zone`. Would the query here be `dns_soa.+` or does `+` do something different?